app/services/year_of_publication_parsing.rb

```diff
@@ -0,0 +1,29 @@
+module YearOfPublicationParsing
```

Comment:
Love this extraction!

Only thing, i would argue that maybe `YearOfPublicationParser` would fit better.

Plus, if this parser will be used only by the `SearchFormParser`, maybe you could namespace these two files so that it's clear they go together (and the `services` folder does not become messy).

Reply:
Oh got a little bit confused with the naming as originally was adding everything under the module for namespacing (without intending a service object), then ended being a service object and didn't realize the name was out of convention. Good catch @your  and @MarcoBgn.

+1 for the namespacing too. Originally I thought about namespacing both under `evidence_hub`, may help to a possible Domain Design gouping in the future (as `evidence_hub`  seems to be a domain concept around different bits of the app).